src/subsocial-query/subsocial/mutation.ts

```diff
@@ -141,54 +142,92 @@ function sendTransaction<Data>(
         address
       )
       danglingNonceResolver = nonceResolver
-      const unsub = await tx.signAndSend(signer, { nonce }, async (result) => {
-        // the result is only tx hash if its using http connection
-        if (typeof result.toHuman() === 'string') {
-          return resolve(result.toString())
-        }
 
-        resolve(result?.txHash?.toString())
-        if (result.status.isInvalid) {
-          txCallbacks?.onError()
-          globalTxCallbacks.onError({
-            summary,
-            address,
-            data,
-          })
-        } else if (result.status.isBroadcast) {
-          txCallbacks?.onBroadcast()
-          globalTxCallbacks.onBroadcast({
-            summary,
-            data,
-            address,
-          })
-        } else if (result.status.isInBlock) {
-          const blockHash = (result.status.toJSON() ?? ({} as any)).inBlock
-          let explorerLink: string | undefined
-          if (networkRpc) {
-            explorerLink = getBlockExplorerBlockInfoLink(networkRpc, blockHash)
+      let usedTx = tx
+      if (proxyToAddress) {
+        usedTx = apis.substrateApi.tx.proxy.proxy(proxyToAddress, null, tx)
+      }
+
+      // signer from talisman and signer from keyring are different
+      // so they need to be handled differently, the one that have 'signPayload' are for talisman signer
+      let account = signer
+      let signerOpt = undefined
+      if ('signPayload' in signer) {
+        account = address
+        signerOpt = signer
+      }
```

Comment:
In sub.id we use one flow for all w/ sigmer prop

Reply:
I still can't find the solution, in sub.id isn't it different because the tx is signed using signer from injected wallet?
in here, it needs to differentiate if the signer is from injected wallet or from keyring